content/cookbook/transactions/offline-transactions.md

```diff
@@ -0,0 +1,388 @@
+} from "@solana/web3.js";
+
+(async () => {
+  // Setup our connection and wallet
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+  const feePayer = Keypair.generate();
+
+  // Fund our wallet with 1 SOL
+  const airdropSignature = await connection.requestAirdrop(
+    feePayer.publicKey,
+    LAMPORTS_PER_SOL,
+  );
+  await connection.confirmTransaction(airdropSignature);
+
+  const nonceAccountAuth = Keypair.generate();
+
+  let nonceAccount = Keypair.generate();
+  console.log(`nonce account: ${nonceAccount.publicKey.toBase58()}`);
+
+  let tx = new Transaction().add(
+    // create nonce account
+    SystemProgram.createAccount({
+      fromPubkey: feePayer.publicKey,
+      newAccountPubkey: nonceAccount.publicKey,
+      lamports:
+        await connection.getMinimumBalanceForRentExemption(
+          NONCE_ACCOUNT_LENGTH,
+        ),
+      space: NONCE_ACCOUNT_LENGTH,
+      programId: SystemProgram.programId,
+    }),
+    // init nonce account
+    SystemProgram.nonceInitialize({
+      noncePubkey: nonceAccount.publicKey, // nonce account pubkey
+      authorizedPubkey: nonceAccountAuth.publicKey, // nonce account authority (for advance and close)
+    }),
+  );
+
+  console.log(
+    `txhash: ${await sendAndConfirmTransaction(connection, tx, [feePayer, nonceAccount])}`,
+  );
+})();
+```
+
+### Get Nonce Account
+
+```typescript filename="get-nonce-account.ts"
+import {
+  clusterApiUrl,
+  Connection,
+  PublicKey,
+  Keypair,
+  NonceAccount,
+} from "@solana/web3.js";
+
+(async () => {
+  // connection
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+
+  const nonceAccountPubkey = new PublicKey(
+    "7H18z3v3rZEoKiwY3kh8DLn9eFT6nFCQ2m4kiC7RZ3a4",
+  );
+
+  let accountInfo = await connection.getAccountInfo(nonceAccountPubkey);
+  let nonceAccount = NonceAccount.fromAccountData(accountInfo.data);
+  console.log(`nonce: ${nonceAccount.nonce}`);
+  console.log(`authority: ${nonceAccount.authorizedPubkey.toBase58()}`);
+  console.log(`fee calculator: ${JSON.stringify(nonceAccount.feeCalculator)}`);
+})();
+```
+
+### Use Nonce Account
+
+```typescript filename="use-nonce-account.ts"
+import {
+  clusterApiUrl,
+  Connection,
+  PublicKey,
+  Keypair,
+  Transaction,
+  SystemProgram,
+  NonceAccount,
+  LAMPORTS_PER_SOL,
+} from "@solana/web3.js";
+import * as bs58 from "bs58";
+
+(async () => {
+  // Setup our connection and wallet
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+  const feePayer = Keypair.generate();
+
+  // Fund our wallet with 1 SOL
+  const airdropSignature = await connection.requestAirdrop(
+    feePayer.publicKey,
+    LAMPORTS_PER_SOL,
+  );
+  await connection.confirmTransaction(airdropSignature);
+
+  // G2FAbFQPFa5qKXCetoFZQEvF9BVvCKbvUZvodpVidnoY
+  const nonceAccountAuth = Keypair.fromSecretKey(
+    bs58.decode(
+      "4NMwxzmYj2uvHuq8xoqhY8RXg63KSVJM1DXkpbmkUY7YQWuoyQgFnnzn6yo3CMnqZasnNPNuAT2TLwQsCaKkUddp",
```

Comment:
Keeping secret keys in source code is bad practice. Maybe use https://github.com/solana-developers/helpers?tab=readme-ov-file#get-a-keypair-from-a-keypair-file instead?

Reply:
Makes Sense

Reply:
Will probably have to revisit this example when code-import is enabled,
since nonceAccountAuth keypair was generated earlier.